### PR TITLE
Add support for arm64

### PR DESCRIPTION
**Pull request by chanchiwai-ray**
- Update `charmcraft.yaml` to support arm64
- Update `check.yaml` to support arm64
- Update `release.yaml` to support arm64
- Removed LMA charms from the test bundle, they are not actively maintained, and can have issues.
- Conditionally set up vm with kvm in the functional test since `Ubuntu_ARM64_4C_16G_01` does not support kvm

### CODE REVIEW

**Review by samuelallan72 — CHANGES_REQUESTED** [review]
This looks pretty good, but please also update release.yaml. :)

File: .github/workflows/check.yaml [hunk 1]
```diff
@@ -26,17 +26,46 @@ jobs:
...
+          python-version: "3.10"
+      - name: Setup Juju environment
+        uses: charmed-kubernetes/actions-operator@main
+        with:
+          provider: "lxd"
+          juju-channel: ${{ matrix.juju-channel }}
+      - name: Install latest tox version
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install tox
```
Comment (rgildein) [inline .github/workflows/check.yaml:62]: This is not needed, since tox is installed via action-operator. We used to re-install specific version.

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -10,20 +10,26 @@ jobs:
...
     secrets: inherit
-
   release:
     needs: check
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.runs-on }}
+    strategy:
+      fail-fast: false
+      matrix:
+        runs-on: [[ubuntu-latest], [Ubuntu_ARM64_4C_16G_01]]
```
Comment (rgildein) [inline .github/workflows/release.yaml:19]: not-blocking: It's just more readable.
```suggestion
        runs-on: 
          - [ubuntu-latest]
          - [Ubuntu_ARM64_4C_16G_01]
```

File: .github/workflows/check.yaml (hunk 1 above)
Comment (chanchiwai-ray) [reply .github/workflows/check.yaml]: Done

File: .github/workflows/release.yaml (hunk 2 above)
Comment (chanchiwai-ray) [reply .github/workflows/release.yaml]: Done

File: src/tests/functional/tests/bundles/base.yaml [hunk 3]
```diff
@@ -5,29 +5,7 @@ applications:
...
-    channel: latest/edge
-  prometheus:
-    charm: prometheus2
-    num_units: 1
-    series: focal
-  grafana:
-    charm: grafana
-    series: focal
-    channel: latest/edge # Temporary fix for 2023.10 release
-    num_units: 1
```
Comment (rgildein) [inline src/tests/functional/tests/bundles/base.yaml:20]: I agree that we should remove the Prometheus and Grafana deployment, but should we at least deploy grafana-agent and check the relations? In grafana-agent we can run `curl localhost:12345/agent/api/v1/metrics/targets` and check status of each target.

Also I do not understand how this is  related with arm64, can you make separate PR for this?

Comment (chanchiwai-ray) [reply src/tests/functional/tests/bundles/base.yaml:20]: grafana arm64 version failed to install due to https://github.com/canonical/charm-prometheus-libvirt-exporter/actions/runs/10346737780/job/28635804387

Comment (chanchiwai-ray) [reply src/tests/functional/tests/bundles/base.yaml:20]: we can test the endpoint from the exporter endpoint itself, there's no need for grafana-agent

Comment (rgildein) [reply src/tests/functional/tests/bundles/base.yaml:20]: It's [recommended](https://discourse.charmhub.io/t/how-to-testing-cos-integrations/14532#tests-for-the-metrics-endpoint-and-scrape-job-4) approach from o11y team, that's why I'm suggesting to do it.

Dropping lot of tests only because of arm64 it's strange approach to me. Those tests can be marked as skipped if they are run on arm64 machine, but not dropping them from normal amd64 tests.

Comment (chanchiwai-ray) [reply src/tests/functional/tests/bundles/base.yaml:20]: > It's [recommended](https://discourse.charmhub.io/t/how-to-testing-cos-integrations/14532#tests-for-the-metrics-endpoint-and-scrape-job-4) approach from o11y team, that's why I'm suggesting to do it.

That's for native COS, right? This is not native COS, this is LMA...

> Dropping lot of tests only because of arm64 it's strange approach to me. Those tests can be marked as skipped if they are run on arm64 machine, but not dropping them from normal amd64 tests.

Sounds reasonable, I can add it back. 👍🏼

Comment (chanchiwai-ray) [reply src/tests/functional/tests/bundles/base.yaml:20]: oh, wait. It's not the test failed, but the install hook failed, so we can't skip test

**Comment by chanchiwai-ray** [comment]
Perhaps it's the same reason why func test for arm64 is failling: https://github.com/canonical/charm-nrpe/pull/193. The func test try to set up lxd container and spin up a vm, but there's no `/dev/kvm`

File: src/tests/functional/tests/bundles/base.yaml (hunk 3 above)
Comment (rgildein) [reply src/tests/functional/tests/bundles/base.yaml:20]: > That's for native COS, right? This is not native COS, this is LMA...

Ah, I did not realize that it's LMA.... you can ignore my comment than.

**Review by rgildein — APPROVED** [review]

**Review by samuelallan72 — APPROVED** [review]
